Shopping cart page

twigmytwig opened the pull request:
ticket NBL-14

twigmytwig committed:
fix: shoping cart interface rebuilt

twigmytwig committed:
fix: shopping cart components deleted

twigmytwig committed:
feat: implemented shopping cart w/ display

twigmytwig committed:
Merge branch 'main' into ShoppingCartPage

abellnoel commented on src/interfaces/shoppingCartInterface.ts:
```diff
@@ -0,0 +1,5 @@
+import { Product } from 'src/interfaces/productInterface';
+
+export interface ShoppingCart{
+    productsInCart: Product[];
```
"productsInCart" may be a little redundant since it's in a ShoppingCart object already. 
```suggestion
    products: Product[];
```

abellnoel commented on src/pages/cart/cart.component.ts:
```diff
@@ -1,14 +1,26 @@
 })
-export class CartComponent {
+export class CartComponent implements OnInit{
+    cartProducts : Product[] = [];
```
Same idea. This will always be in the context of the CartComponent. 
```suggestion
    products: Product[] = [];
```

abellnoel commented on src/services/shopping-cart.service.ts:
```diff
@@ -7,8 +8,17 @@ import { Product } from 'src/interfaces/productInterface';
 export class ShoppingCartService {
 
   //constructor() { }
```
removing commented code
```suggestion
```

twigmytwig committed:
feat: add cart component

I added the cart component to the nav bar and it keeps track of your cart
without you having to go to the cart page.

twigmytwig committed:
Merge branch 'ShoppingCartPage' of https://github.com/abellnoel/nooblion-commerce into ShoppingCartPage

abellnoel commented on src/services/shopping-cart.service.ts:
```diff
@@ -7,8 +8,17 @@ import { Product } from 'src/interfaces/productInterface';
 
-  onAddToCartClicked(productName: string){
-    alert('You have added ' + productName + ' To your cart!');
+   onAddToCartClicked(product: Product){
```
Sounds more like an event listener name than an action. We could probably name this something more general like `addProductToCart`.  
```suggestion
   addProductToCart(product: Product){
```

abellnoel commented on src/components/shopping-cart-drop-down/shopping-cart-drop-down.component.html:
```diff
@@ -0,0 +1,27 @@
+    <div class="dropdown">
+    <ul><button  class="btn btn-link dropdown-toggle" data-bs-toggle="dropdown" type="button" href="" id="cart">
+
+        <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-cart" viewBox="0 0 16 16">
```
We could try using bootstrap icons for these instead: https://icons.getbootstrap.com/

twigmytwig replied:
what I used is a bootstrap icon me finks

abellnoel reviewed (changes requested):
Dope stuff, just left a few naming suggestions

twigmytwig committed:
testing for cart total

twigmytwig committed:
fix: cart total is now working properly

twigmytwig committed:
Update src/interfaces/shoppingCartInterface.ts

Co-authored-by: Noel Abellanosa <59884349+abellnoel@users.noreply.github.com>

twigmytwig committed:
Update src/pages/cart/cart.component.ts

Co-authored-by: Noel Abellanosa <59884349+abellnoel@users.noreply.github.com>

twigmytwig committed:
Update src/services/shopping-cart.service.ts

Co-authored-by: Noel Abellanosa <59884349+abellnoel@users.noreply.github.com>

twigmytwig committed:
Update src/services/shopping-cart.service.ts

Co-authored-by: Noel Abellanosa <59884349+abellnoel@users.noreply.github.com>

twigmytwig committed:
fix: changed naming conventions

twigmytwig committed:
Merge branch 'main' into ShoppingCartPage

twigmytwig committed:
optimization of one variable.....

twigmytwig committed:
Merge branch 'ShoppingCartPage' of https://github.com/abellnoel/nooblion-commerce into ShoppingCartPage
